### ISSUE TITLE
HW3 Angular Test Grading

### ISSUE THREAD

**Issue by snumath** [swsnu/swppfall2018#110]
질문 답해주시느라 수고 많으십니다. 숙제 3 질문이니 나중에 답장해주셔도 괜찮습니다.
숙제 3에서 Angular Test에 15점이 할당돼있습니다.
아직 Coverage가 어떤 방식으로 산출되는지 잘 모르겠지만 coverage가 높을수록수록 좋은 Test라고 배웠습니다.
과제 3은 과제 2에서 구현했던 코드를 구현하는 것인데요, 과제 2에서 만들었던 코드가 엉망진창이라 테스트가 힘들 때에도 과제2의 코드를 변경하면 안 되는건가요?
만약 코드를 변경해도 된다면 얼마나 변경해도 되나요? (극단적으로 모든 메인 컴퍼넌트를 제외한 모든 컴퍼넌트를 날려버린다면 coverage 100%는 쉽게 될텐데, 당연히 이를 허용하실거 같진 않습니다.)

**Comment by sanha** [comment]
네, 말씀하신 것 처럼 feature를 줄여 coverage를 높이는 것을 방지하기 위해 (readme에도 나와있듯이) 코드 수정을 허용하지 않습니다.
다만 테스트가 어려운 코드를 작성했을 수 있고, Angular test 자체가 100%의 coverage를 보이기 어려운 만큼 reasonable한 정도의 coverage만 보이면 만점을 드리는 방향으로 생각하고 있습니다.
특별히 code상의 문제로 일정 이상의 coverage를 달성하기 어려운 경우 해당 내용을 메일로 보내주시면 검토 후 채점에 반영하도록 하겠습니다. (다만 이러한 메일은 HW 3의 제출 듀가 지난 후 한번만 일괄적으로 검토하겠습니다.)

**Comment by snumath** [comment]
감사합니다. resolved label 달아주세요!

**Comment by tldn0718** [comment]
그렇다면 과제2 때 제출한 코드에 버그가 있거나, 구현 못 한 feature가 있거나, 하드코딩한 부분이 있는 경우에도 코드를 수정해서는 안 되는건가요? feature을 줄이는 꼼수가 아니라 버그 수정, feature 구현 등의 상식적인 수준에서의 코드 개선도 불허하는건가요?!

**Comment by sanha** [comment]
네, 기본적으로 HW 2에 같이 제출하셨어야 할 테스팅이나 강의 일정 관계로 미루어진 것이므로, 그 코드를 기준으로 작성해주셔야 합니다. 다만 feature들이 잘 동작하는지를 다시 테스트해서 감점하지는 않습니다.
극단적인 예를 들어, 아무것도 구현하지 않고 HW 2를 제출했다면 구현 점수는 0점이지만 testing은 쉽게 coverage 100% 를 달성할 수 있습니다. 이는 구현과 테스팅을 한꺼번에 제출한 경우에도 같습니다.
